spacemesh/v1/post_types.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+package spacemesh.v1;
+option go_package = "github.com/spacemeshos/api/release/go/spacemesh/v1";
+
+import "spacemesh/v1/types.proto";
+
+// NodeRequest is a request sent from the node to the post service.
+message NodeRequest {
+  oneof kind {
+    // StateRequest state = 1;
+    // MetadataRequest metadata = 2;
+    // InitRequest init = 3;
+    // ResizeRequest resize = 4;
+    GenProofRequest gen_proof = 5;
+  }
+}
+
+// ServiceResponse is a response sent from the post service to the node.
+message ServiceResponse {
+  oneof kind {
+    // StateResponse state = 1;
+    // MetadataResponse metadata = 2;
+    // InitResponse init = 3;
+    // ResizeResponse resize = 4;
+    GenProofResponse gen_proof = 5;
+  }
+}
+
+message GenProofRequest { bytes challenge = 1; }
+
+message Proof {
+  uint32 nonce = 1;
+  bytes indices = 2;
+  uint64 pow = 3;
+}
+
+message ProofMetadata {
+  bytes challenge = 1;
+  SmesherId node_id = 2;
+  ActivationId commitment_atx_id = 3;
+  uint32 num_units = 4;
+  uint64 labels_per_unit = 5;
+}
+
+enum GenProofStatus {
```

Comment:
What about proof generation in progress? Don't we want to be able to report the progress?

Reply:
We are currently figuring this out as we go. That's why the PR is still a draft 🙂. The current idea is that the node is stateless and in regular intervals sends the same "GenerateProof" request to all PoST services connected. These requests are idempotent, so if a service receives a request with the same challenge twice it just responds OK and continues to generate the proof.

At the moment we also don't have any progress information about proof generation and we want to keep the change simple to be able to deliver earlier.

Reply:
I think it's important to be able to report the status of the proof generation while in progress, either via "push" (PoST service reports back with status periodically) or "pull" (node checks in with PoST service periodically). Proof generation takes hours and it's important the the user be able to see what's going on. See https://github.com/spacemeshos/go-spacemesh/issues/4742.

Reply:
Thats out of scope for this in my opinion. There is no easy way to know the current status of the proof generation.

The post service can keep track of how far into reading all data from disk it is, but this cannot be extrapolated to how long it will take for the proof generation to be finished. Because

- it takes x amount of time to read the first 50% of data; this does not mean that it takes the same amount of time to read the remaining data - depending on the disk and file system this could be quite different
- Just because it read 50% of the data doesn't mean the proof generation is done - it could take 1 pass or it could take 10 passes (depending on number of nonces per pass, etc.) to generate a proof. It's probabilistic there is no reliable way to tell how much more time it will take until a proof will be available.

We could report back to the node how much data has been read from disk already and which pass over the data it is, but how would we display that information to the user?

I would prefer to first discuss on how we measure progress, how we display it to the user and then design the interface around that and not the other way round 🙂

Reply:
Additionally the post service logs progress about proof generation via `os.Stderr` and if used in standalone mode that output will be captured and logged by the node as debug logs.

If you think this should be info logs instead I can change that 🙂

Reply:
@fasmat 
>I would prefer to first discuss on how we measure progress, how we display it to the user and then design the interface around that and not the other way round 🙂

100% agree. Measuring POST proof generation progress is not trivial.

Note that this is not the final version of this API and we can iterate on it, adding more features as we go.